nav2_costmap_2d/plugins/inflation_layer.cpp

```diff
@@ -113,6 +120,7 @@ InflationLayer::onInitialize()
 void
 InflationLayer::matchSize()
 {
+  std::lock_guard<Costmap2D::mutex_t> guard(*getMutex());
```

Comment:
My understanding of the mutex in the inflation layer is that it is for locking in the inflation resource (see the doxygen comment on it) and the previous implementation was only locking the update / values in the inflation cost data structure. It feels like locking it also for the parameters is abusing the role here a little.

 Would it make more sense to have these variables as atomics instead?

Reply:
Good point. Maybe its better to use another mutex then -- or do you think its OK that we're kind of abusing this mutex? Its not really for the same thing anymore. At bare minimum, the description of the mutex needs updating.

Reply:
For InflationLayer, standard mutex was chosen to cover `inflation_radius_` and `need_reinflation_` variables shared usage by `matchSize()`, `updateBounds()`, `onFootprintChanged()` and `dynamicParametersCallback()`. However, introducing of new lock for them will cause to have 2 mutexes in `updateCosts()` + `computeCaches()` or `dynamicParametersCallback()` which is also accessing `cell_inflation_radius_` parameter used by already covered by standard mutex routines. Therefore, I've suggested to use one mutex for everything for simplicity.

Reply:
OK, I think that's fine